packages/legend-application-studio/src/components/editor/editor-group/service-editor/ServiceExecutionQueryEditor.tsx

```diff
@@ -249,14 +252,33 @@ export const ServiceExecutionQueryEditor = observer(
           };
         }
       } else {
+        const currentWorkSpaceId =
```

Comment:
is there any other places where we call legend query that we can leverage this or no?

If so maybe we should calculate whether a workspace contains snapshot in the initial fetching of versions,etc.